src/lib/feature-flags/flags.ts

```diff
@@ -0,0 +1,87 @@
+import window from 'global/window';
+
+export type FeatureFlagKey =
+  | 'channelsApp'
+  | 'enableDevPanel'
+  | 'resetPasswordPage'
+  | 'enableRewards'
+  | 'enableMatrix'
+  | 'allowEmailRegistration'
+  | 'verboseLogging'
+  | 'allowEditPrimaryZID'
+  | 'enableReadReceiptPreferences'
+  | 'enableUserSettings'
+  | 'enableTimerLogs'
+  | 'enableChannels'
+  | 'enableCollapseableMenu'
+  | 'enableMeows'
+  | 'enableTokenGatedChat'
+  | 'enableNotificationsApp'
+  | 'enableNotificationsReadStatus'
+  | 'enableLoadMore'
+  | 'enableComments'
+  | 'enableFeedApp'
+  | 'enableLinkedAccounts'
+  | 'enableZeroWalletSigning'
+  | 'enableFeedChat';
+
+export interface FeatureFlagConfig {
+  defaultValue: boolean;
+  locked?: boolean;
+  allowedUserIds?: string[];
+}
+
+export type FeatureFlagDefinitions = Record<FeatureFlagKey, FeatureFlagConfig>;
+
+export type FeatureFlagValues = Record<FeatureFlagKey, boolean>;
+
+/**
+ * Bypass the store dependency for the current user ID if we're in a test environment.
+ */
+let getCurrentUserId = () => null;
+export const setGetCurrentUserId = (fn: () => string | null) => {
+  getCurrentUserId = fn;
+};
+if (process.env.NODE_ENV !== 'test') {
+  (async () => {
+    const { store } = await import('../../store');
+    getCurrentUserId = () => store.getState().authentication?.user?.data?.id ?? null;
+  })();
+}
```

Comment:
We don't have any tests around feature flags themselves (outside of the FeatureFlag component), so I'm bypassing this saga dependency when we're in tests.
This means we won't be able to write tests around `allowedUserIds` as the current user id will always be null, but that seems like a reasonable tradeoff instead of mocking this in all tests that use it.

Reply:
For reference, there were 50 failed tests due to pulling in the store